tasks/main.yml

```diff
@@ -172,11 +175,23 @@
         key: '{{ user.pubkey }}'
         key_options: '{{ user.options | default(omit) }}'
         state: "{{ user.key_state | default(omit) }}"
+        exclusive: "{{ user.pubkey_exclusive | default(pubkey_exclusive_default) }}"
```

Comment:
I wonder if I read this right. This line sets exclusive for the single key case, so if `crap_key` exists it will get cleared.

However, the multikey case does not set exclusive so `crap_key` will stay there. Any reson not to enforce this for both?

Reply:
It would be kind of interesting to know how this works for e.g. scenario where user has

```
ssh-rsa ansible_controlled_hash_key_1
ssh-rsa ansible_controlled_hash_key_2
ssh-rsa crap_key
```

they want to e.g. change the 2nd key

Reply:
It would work craptastic, you will have a bad time, the role will change after each run.

First task: Will ¿always? append keys if one tires to update a key if exclusive is not true. So it is hard to control a users key without exclusive, which limits the user to one key.

Second task: You need to manage existing and absent keys. You can't use exclusive since it is a while loop.

I consider that both tasks leaves things to be desired, non of the functions makes sure that "these multiple keys are present but no other key".

Reply:
Regarding the first task - 

https://docs.ansible.com/ansible/latest/collections/ansible/posix/authorized_key_module.html#parameter-exclusive

says _Multiple keys can be specified in a single key string value by separating them by newlines._

Tried this just now, seems to work:

```
$ git diff
-    pubkey: "ssh-rsa KEY1\nssh-rsa KEY2"
+    pubkey: "ssh-rsa KEY1\nssh-rsa NEW_KEY2"
$ ansible-playbook -i inv -l limit site.yml -CD -e pubkey_exclusive_default=True --tags users
...
 ssh-rsa KEY1
-ssh-rsa KEY2
-ssh-rsa crapkey
+ssh-rsa NEW_KEY2 
```

For me the primary thing would be that this method, which is in the official Ansible documentation, works. Not sure why one would remove the official method from the readme though, would just removing the word "trick" suffice?

In any case IIUC `pubkey` is retained and `pubkeys` is added alongside as an alternative. Works for me.

Reply:
So the exclusive was added as an after thought, since I thought it was a good feature to add. I did not consider that one could manage multiple keys *with* exclusive. 

I don't think the ansibles official method for managing multiple keys is any good. \< insert skinner meme - am I wrong > . So if you think that `pubkey` is preferred method to managing multiple keys, IMO the README.md should not be updated, and the `pubkeys` task should not be added.

The reason why I added `pubkeys` was because I did not want to manage multiple keys with `newline`.

Reply:
[Inserted](https://media.giphy.com/media/V9gjxvLnSSdA4/giphy.gif)

Yeah, I'm getting old. And on my olden days I seem to be more lenient towards both the dreaded Rosso's table of grated vegetables as well as abiding to the vendor documentation.

So even if it's a bit hairy and may result in ultra long rows, that's how I would prefer to use the role, as long as upstream Ansible documents it as such.

Reply:
So anymore review things? Should we close this PR or merge it? 

The main reason for the PR was because a colleague mange to lock himself out of his instance, when he wanted to update his ssh key.

Reply:
In summary

- Joining the lists is a definite improvement for future proofing the role
- Removing ssh_args is an improvement
- IMO the exclusive is also an improvement even if it can have caveats for users who would blindly enable the flag.

For the actual multikey stuff, if this was my PR, I would just readme `s/trick/Ansible documented method [link]/` and try not to introduce two competing methods for doing the same thing.

But this is not my PR. Since `pubkey` support is not removed, I'll approve and outsource the decision to you.